### PR TITLE
Adding k8s-infra-kops-scale-tests bucket with public read

### PR DESCRIPTION
**Pull request by pkprzekwas**
Creating dedicated bucket with public read access for kops scale tests (as discussed [here](https://kubernetes.slack.com/archives/CCK68P2Q2/p1691083904395479?thread_ts=1689009672.882279&cid=CCK68P2Q2))

/assign @ameukam @dims 
/cc @prateekgogia

### CODE REVIEW

File: infra/aws/terraform/s3/k8s-infra-e2e-boskos-scale-001/providers.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+*/
+
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+  }
+}
```
Comment (ameukam) [inline infra/aws/terraform/s3/k8s-infra-e2e-boskos-scale-001/providers.tf:24]: @pkprzekwas   all resources created here should live in a remote backend. Which backend is used ?

Comment (pkprzekwas) [reply infra/aws/terraform/s3/k8s-infra-e2e-boskos-scale-001/providers.tf:24]: Ah, fair point. We don't have any bucket for TF state on this account yet. Let me create and use it.

Comment (pkprzekwas) [reply infra/aws/terraform/s3/k8s-infra-e2e-boskos-scale-001/providers.tf]: @ameukam added configuration for remote tf state.

**Comment by ameukam** [comment]
/approve

Leave to @prateekgogia

File: infra/aws/terraform/s3/k8s-infra-e2e-boskos-scale-001/.gitignore [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Local .terraform directories
+**/.terraform/*
+
```
Comment (ameukam) [inline infra/aws/terraform/s3/k8s-infra-e2e-boskos-scale-001/.gitignore:3]: @pkprzekwas suggestion: we should probably move this .gitignore file to `../infra/aws/terraform` so all the sub-repos inherit this config.

Comment (pkprzekwas) [reply infra/aws/terraform/s3/k8s-infra-e2e-boskos-scale-001/.gitignore:3]: We can think about having some generic ignore at the top level of the terraform directory. But that should be a dedicated PR that would also remove other ignores.

**Comment by prateekgogia** [comment]
LGTM `(Don't have permissions to approve)`

**Pull request merged**